src/components/Autocomplete/components/ComboBox/tests/ComboBox.test.tsx

```diff
@@ -588,3 +542,8 @@ function renderTextField() {
 function renderNodeWithId() {
   return <div id="CustomNode" />;
 }
+
+function dispatchKeyup(key: Key) {
+  const event: KeyboardEventInit & {keyCode: Key} = {keyCode: key};
```

Comment:
I'm not well versed (or opinionated) on test architecture, but I don't know if we need to go through actually dispatching a real event here. Triggering the handler is another option:

```suggestion
function dispatchKeyup(wrapper: ReactWrapper, key: Key) {
  const listener = wrapper.findWhere((listener) => {
    return listener.prop('keyCode') === key;
  });

  trigger(listener, 'handler', {keyCode: key});
}
```

and then in the tests you'd just pass in the `comboBox` wrapper as the first arg, or `comboBox.find(Popover)` in the case of the `Key.Escape` test since the listener for that is in the popover component. But imo that test should be 🔥 because it's testing functionality of a parent that already tests that.

Reply:
I maybe totally off but the issue I see with trigger is that it'll dispatch `onSelect` for example however when a key is pressed the `ComboBox` has it's own internal logic that handle the keyup and then call `onSelect` with the result of ComboBox's internal logic

Reply:
`trigger` will just look for a corresponding handler passed through props, and trigger that handler, so it doesn't feel useful for us in this case. Dispatching the event feels like the way to go from my perspective